numpy/lib/_info.py

```diff
@@ -0,0 +1,384 @@
+        raise ValueError(
+            f'Device not understood. Only "cpu" is allowed, but received: {device}'
+        )
+    return {
+        "real floating": float64,
+        "complex floating": complex128,
+        "integral": int64,
+        "indexing": int64,
+    }
+
+
+def dtypes(
+    *,
+    device: Optional[str] = None,
+    kind: Optional[Union[str, Tuple[str, ...]]] = None,
+) -> DataTypes:
+    """
+    The array API data types supported by NumPy.
+
+    Note that this function only returns data types that are defined by the
+    array API.
+
+    Parameters
+    ----------
+    device : str, optional
+        The device to get the data types for. For NumPy, only ``'cpu'`` is
+        allowed.
+    kind : str or tuple of str, optional
+        The kind of data types to return. If ``None``, all data types are
+        returned. If a string, only data types of that kind are returned. If a
+        tuple, a dictionary containing the union of the given kinds is
+        returned. The following kinds are supported:
+
+            - ``'bool'``: boolean data types (e.g., ``bool``).
+            - ``'signed integer'``: signed integer data types (e.g., ``int8``,
+              ``int16``, ``int32``, ``int64``).
+            - ``'unsigned integer'``: unsigned integer data types (e.g.,
+              ``uint8``, ``uint16``, ``uint32``, ``uint64``).
+            - ``'integral'``: integer data types. Shorthand for ``('signed
+              integer', 'unsigned integer')``.
+            - ``'real floating'``: real-valued floating-point data types
+              (e.g., ``float32``, ``float64``).
+            - ``'complex floating'``: complex floating-point data types (e.g.,
+              ``complex64``, ``complex128``).
+            - ``'numeric'``: numeric data types. Shorthand for ``('integral',
+              'real floating', 'complex floating')``.
+
+    Returns
+    -------
+    dtypes : DataTypes
+        A dictionary mapping the names of data types to the corresponding
+        NumPy data types.
+
+    See Also
+    --------
+    capabilities, default_device, default_dtypes, devices
+
+    Examples
+    --------
+    >>> info = np.__array_namespace_info__()
+    >>> info.dtypes(kind='signed integer')
+    {'int8': numpy.int8,
+     'int16': numpy.int16,
+     'int32': numpy.int32,
+     'int64': numpy.int64}
+
+    """
+    if device not in ["cpu", None]:
+        raise ValueError(
+            f'Device not understood. Only "cpu" is allowed, but received: {device}'
+        )
+    if kind is None:
+        return {
+            "bool": bool,
+            "int8": int8,
+            "int16": int16,
+            "int32": int32,
+            "int64": int64,
+            "uint8": uint8,
+            "uint16": uint16,
+            "uint32": uint32,
+            "uint64": uint64,
+            "float32": float32,
+            "float64": float64,
+            "complex64": complex64,
+            "complex128": complex128,
```

Comment:
Nitpick: maybe instead of repeating contents of these dicts, let's keep `signed integer`, `unsigned integer`, `bool`, `real`, `complex` dicts, and define the rest of them by merging existing ones: `integral = signed_integer | unsigned_integer` etc.